platform/platform.go

```diff
@@ -0,0 +1,279 @@
+	"github.com/kiegroup/kogito-serverless-operator/utils"
+	coordination "k8s.io/api/coordination/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"os"
+	ctrl "sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+)
+
+const (
+	// DefaultPlatformName is the standard name used for the platform.
+	DefaultPlatformName = "kogito-serverless-platform"
+
+	OperatorWatchNamespaceEnvVariable = "WATCH_NAMESPACE"
+	operatorNamespaceEnvVariable      = "NAMESPACE"
+	operatorPodNameEnvVariable        = "POD_NAME"
+)
+
+// Copied from https://github.com/kubernetes/enhancements/tree/master/keps/sig-cluster-lifecycle/generic/1755-communicating-a-local-registry
+// LocalRegistryHostingV1 describes a local registry that developer tools can
+// connect to. A local registry allows clients to load images into the local
+// cluster by pushing to this registry.
+type LocalRegistryHostingV1 struct {
+	// Host documents the host (hostname and port) of the registry, as seen from
+	// outside the cluster.
+	//
+	// This is the registry host that tools outside the cluster should push images
+	// to.
+	Host string `yaml:"host,omitempty"`
+
+	// HostFromClusterNetwork documents the host (hostname and port) of the
+	// registry, as seen from networking inside the container pods.
+	//
+	// This is the registry host that tools running on pods inside the cluster
+	// should push images to. If not set, then tools inside the cluster should
+	// assume the local registry is not available to them.
+	HostFromClusterNetwork string `yaml:"hostFromClusterNetwork,omitempty"`
+
+	// HostFromContainerRuntime documents the host (hostname and port) of the
+	// registry, as seen from the cluster's container runtime.
+	//
+	// When tools apply Kubernetes objects to the cluster, this host should be
+	// used for image name fields. If not set, users of this field should use the
+	// value of Host instead.
+	//
+	// Note that it doesn't make sense semantically to define this field, but not
+	// define Host or HostFromClusterNetwork. That would imply a way to pull
+	// images without a way to push images.
+	HostFromContainerRuntime string `yaml:"hostFromContainerRuntime,omitempty"`
+
+	// Help contains a URL pointing to documentation for users on how to set
+	// up and configure a local registry.
+	//
+	// Tools can use this to nudge users to enable the registry. When possible,
+	// the writer should use as permanent a URL as possible to prevent drift
+	// (e.g., a version control SHA).
+	//
+	// When image pushes to a registry host specified in one of the other fields
+	// fail, the tool should display this help URL to the user. The help URL
+	// should contain instructions on how to diagnose broken or misconfigured
+	// registries.
+	Help string `yaml:"help,omitempty"`
+}
+
+const OperatorLockName = "kogito-serverless-lock"
+
+var OperatorImage string
+
+// IsCurrentOperatorGlobal returns true if the operator is configured to watch all namespaces.
```

Comment:
will we allow the operator to run as namespaced?

Reply:
No at the moment but the code is ready thanks to approach that Camel-k is using and that we ported here into the operator.